### DISCUSSION TITLE
vel_update

### DISCUSSION THREAD

**Discussion by YSatShinshu** [rodralez/NaveGo#87] [Q&A] [answered]
Dear NaveGo-Team, thank you for your very helpful work.

I have a question about function "vel_update".

Why do you not multiply the Coriolis acceleration by the velocity in your velocity update formula?

Here is the location (vel_update.m, line 44-46)

> coriolis = (omega_en_n + 2 * omega_ie_n);   % Coriolis 
> 
> fn_c = fn - coriolis + gn;                  % Corrected specific force in nav-frame

I'm sorry for my lack of study, but it seems to be multiplied in the reference, so please tell me.

Thank you.

**Comment by rodralez** [comment, marked as answer]
Hello,

Thank you for pointing out this BIG bug.

I had to modified the following functions:

ins-gnss/ins_gnss.m
ins/coriolis.m
ins/earth_rate.m
ins/transport_rate.m
ins/vel_update.m
simulation/acc_gen.m
simulation/gyro_gen.m

The transport and Earth rates were incorrectly defined in NaveGo as well.

Since these turn rates and Corolioris can be neglected in navigation systems that are based on tactical-grade sensors and inferior categories, NaveGo's examples are unaffected by these changes.

Kind regards.

**Reply by YSatShinshu** [reply]
Thanks for the fix!

**Reply by rodralez** [reply]
I will appreciate if you can check if the following functions are right:

ins/earth_rate.m
ins/transport_rate.m
ins/vel_update.m

Kind regards.

**Reply by YSatShinshu** [reply]
As far as I can confirm, there is no problem.

On a different note, NaveGo uses the previous value for the speed multiplied by Coriolis. In the examples I know, it uses the velocity derived by interpolation or extrapolation, and I'm not sure why. Do you think it is OK to use the previous value?  Please let me know your opinion.

**Reply by rodralez** [reply]
I have tested velocity interpolation for updating the velocity (vel_update.m) but NaveGo performance does not change, at least using tactical- and automotive-grade IMUs.

**Reply by venkat057** [reply]
> As far as I can confirm, there is no problem.
> 
> On a different note, NaveGo uses the previous value for the speed multiplied by Coriolis. In the examples I know, it uses the velocity derived by interpolation or extrapolation, and I'm not sure why. Do you think it is OK to use the previous value? Please let me know your opinion.

Hi,
    Velocity derived by interpolation or extrapolation gives better accuracy only when vehicle dynamics are high like vehicle speed is high. Since in real-time by the time kf compute the position, velocity the vehicle is already moved to a new position, velocity. So the kf outputs are delayed/old by kf execution time.
Can you send me link of this Velocity derived by interpolation or extrapolation example.

**Reply by YSatShinshu** [reply]
Hello,

Sorry, because the example I have is a paid one, I send you a link to the sales page. I use "INS Toolbox 3.0@MATLAB"
GPsoft LCC. 
https://gpsoftnav.com/products/

> Velocity derived by interpolation or extrapolation gives better accuracy only when vehicle dynamics are high like vehicle speed is high. Since in real-time by the time kf compute the position, velocity the vehicle is already moved to a new position, velocity. So the kf outputs are delayed/old by kf execution time.

Thanks for giving me advice. Is this generally known? If you know any references, please tell me.

Best regards.

**Reply by rodralez** [reply]
Velocity update code is based on Eq. 5.53 and 5.54, page 179, from Paul D. Groves, _Principles of GNSS, Inertial, and Multisensor Integrated Navigation Systems. Second Edition_. I have realized that previous gravity, previous Transport rate and previous Earth rate should be used instead of current values. This error has to be fixed.
